User story issue#32

DasDeveloper opened the pull request:
Added the frontend for job listing and for job application.

Added the backend for job listing and for job application.

Added tests for the frontend.

Added tests for the backend.

Added jobRoutes.

DasDeveloper committed:
Created job page

DasDeveloper committed:
Changes for JobApplication page

DasDeveloper committed:
Created the routes for jobs and added the jobs model

DasDeveloper committed:
Added functionality for getJobs()

DasDeveloper committed:
Changed the job application page for the user

DasDeveloper committed:
Merge branch 'dev' into UserStoryIssue#32

DasDeveloper committed:
JobListing Page

DasDeveloper committed:
Fixed Job Listing page

DasDeveloper committed:
Changes

DasDeveloper committed:
Update App.js

DasDeveloper committed:
Merge branch 'dev' into UserStoryIssue#32

DasDeveloper committed:
Added the routes for jobListing and jobApplication again

DasDeveloper committed:
Adjusted the routes

DasDeveloper committed:
Changed the get request for Job in the jobRoutes.js file

DasDeveloper committed:
Backend for jobListing works

DasDeveloper committed:
Display all jobs in JobListing page

DasDeveloper committed:
Adjusted css for jobListing

DasDeveloper committed:
Merge branch 'dev' into UserStoryIssue#32

DasDeveloper committed:
Adjusted link between a specific job and the job application page

DasDeveloper committed:
Removed debug css for jobListing and adjusted the preferences tab

DasDeveloper committed:
Update JobListing.jsx

DasDeveloper committed:
Fixed the job application

Displays information about the job selected in the jobApplication page

DasDeveloper committed:
Added a route to get all the jobs applied for a specific user

DasDeveloper committed:
Added functionality to jobApplication page

When a user applies for a job, it checks if the user has already applied for it. If so, the page is redirected.

DasDeveloper committed:
Merge branch 'dev' into UserStoryIssue#32

DasDeveloper committed:
Fixed the jobApplication submit functionality

DasDeveloper committed:
Fixed changes

DasDeveloper committed:
Created tests for the frontend and the backend for jobListing and jobApplication

DasDeveloper committed:
Fixed some css for jobApplication and jobListing

Aboudik1452 commented on src/client/src/components/JobListing.jsx:
```diff
@@ -0,0 +1,64 @@
+                        <p>Salary: {job.salary}</p>
+                        <p>Company: {job.company}</p>
+                        <p>Category: {job.category}</p>
+                        <Link to = {`String text ${job.job_id}`} state = {{jobState:job}}>
```
Is `String text` necessary? It is displayed on the address bar
![image](https://user-images.githubusercontent.com/78051064/219801883-3c1f4193-c1f0-40ff-9d75-0ebb29498a95.png)

DasDeveloper replied:
You're right, it was a typo thanks for catching
![Screen Shot 2023-02-17 at 5 33 42 PM](https://user-images.githubusercontent.com/14958097/219808384-18919b96-b2ee-4e4b-b489-7cbbd088857a.png)

Aboudik1452 commented on src/client/src/css/jobListing.css:
```diff
@@ -0,0 +1,89 @@
+
+.Container{
+
+    display: flex;
```
While the page is responsive, it looks messy on a smaller device. Open Inspect on Chrome/Firefox to check this, and toggle Device Preview.

![image](https://user-images.githubusercontent.com/78051064/219803612-cfc31c4b-8d91-4800-86bb-2f1ff8b80632.png)

DasDeveloper replied:
Made the pages responsive for smaller devices.
![Screen Shot 2023-02-17 at 7 06 40 PM](https://user-images.githubusercontent.com/14958097/219819252-1863aea3-03e4-4323-b48e-8471aedaee68.png)
For job listing and job application

Aboudik1452 commented on src/client/src/test/JobPosting.test.js:
```diff
@@ -0,0 +1,18 @@
+test("jobListing page renders properly", ()=>{
+    render(<JobPosting/>);
+
+    const fnameTextboxElement = screen.getByRole('textbok', {name:'fname'});
```
This test failed on my system for this reason:
`TestingLibraryElementError: Unable to find an accessible element with the role "textbok" and name "fname"`

Aboudik1452 replied:
This test failed on my system for this reason:
`TestingLibraryElementError: Unable to find an accessible element with the role "textbok" and name "fname"`

DasDeveloper committed:
Fixed the link between the joblisting component and the job application component

DasDeveloper commented on src/client/src/css/jobListing.css:
```diff
@@ -0,0 +1,89 @@
+
+
+
+.jobPosts .jobPost{
+
+    border: 2px grey solid;
+    margin: 5px;
+    display: block;
+    width: 80%;
+    height: 240px;
+}
+
+.jobPosts .jobPost button{
+    background-color: #19718d;
+    display: block;
+    width: 100px;
+    border-radius: 20px;
+    margin: 15px;
+    text-decoration: none;
+}
```
This should be fixed by changing the height selector to auto.

![Screen Shot 2023-02-17 at 5 42 29 PM](https://user-images.githubusercontent.com/14958097/219811365-25e4de20-7afc-425d-9e4b-6a9d466235b6.png)

Aboudik1452 commented on src/server/tests/josbApplication.test.js:
```diff
@@ -0,0 +1,44 @@
+        it("it successfully adds a job id to the jobsApplied attribute", async ()=>{
+
+            await request(app)
+                .post("/users/63edb27d0e77e161a004824c/jobsApplied", {
+                    userId: '63edb27d0e77e161a004824c',
+                    jobId: '2142412'
+                })
```
On my system, both parameters are not being passed through correctly, I get `undefined`.

DasDeveloper replied:
That test passes on my end.
![Screen Shot 2023-02-17 at 7 29 41 PM](https://user-images.githubusercontent.com/14958097/219821075-f2a1020a-aa2e-497d-a92f-f4f30b2d1e33.png)

Aboudik1452 commented on src/server/routes.rest:
```diff
@@ -0,0 +1,20 @@
+
+
+GET http://localhost:9000/jobs
+
+
+###
+
+GET http://localhost:9000/jobs/1234131231242
+
+###
+
+GET http://localhost:9000/users/63edb27d0e77e161a004824c/jobsApplied
+
+
+###
+
+POST http://localhost:9000/users/63edb27d0e77e161a004824c/jobsApplied
+
+###
+
```
What is this file used for?

DasDeveloper replied:
This is used for an extension. The extensions allows us to send requests just like postman. I'll put it in my gitignore file.

Aboudik1452 commented on src/server/models/user.js:
```diff
@@ -44,4 +44,12 @@ const UserSchema = new mongoose.Schema({
 const User = mongoose.model("Users", UserSchema);
 
 
+// User.create({
+//     firstname: "Bob",
+//     lastname: "Test",
+//     email: "bobTest@hotmail.com",
+//     password: "122121212",
+//     jobsApplied: [1234131231242]
+// })
```
We can re-use sample data as test data for running tests.
You can remove this from the file.

Aboudik1452 commented on src/server/controllers/usersController.js:
```diff
@@ -92,13 +93,53 @@ const getUserInfo = async (req, res) => {
+      jobsApplied: req.body.jobId
+    }
+  }
+  // function(err, result) {
+  //   if (err) {
+  //     res.send(err);
+  //   } else {
+  //     res.send(result);
+  //   }
+  // })
```
You can remove this from the file

Aboudik1452 commented on src/server/models/Job.js:
```diff
@@ -29,4 +29,14 @@ const jobsSchema = new mongoose.Schema({
 
 const Jobs = mongoose.model("Jobs", jobsSchema);
 
+// Jobs.create({
+//     job_id:12341312342323,
+//     description: "test2",
+//     salary: 123,
+//     company: "testCompany2",
+//     category:"Software",
+//     title: "testTitle2"
+// })
```
You can remove this, and re-use it later for testing.

DasDeveloper committed:
Issue with button placement

Fixed an issue where the button would overlap with the border with another job post.

Joejoe1452 reviewed (changes requested):
Good work @DasDeveloper!

Please look at the issues mentioned and check that your functionality works. It's possible that these could be due to how my environment is setup. I will await input from the other reviewers of this pull request.

DasDeveloper committed:
Update .gitignore

DasDeveloper committed:
Made the josListing page responsive for smaller devices

DasDeveloper committed:
Made the job Application page responsive for smaller devices

DasDeveloper committed:
Removed useless comments

DasDeveloper committed:
Fixed some issues

imran1289-ah reviewed (approved):
it looks good overall. i guess we are using some sort of a predefined id to make the request to the backend ?We will probably have to change this in the future and have it fetch the id of the logged in recruiter.

DasDeveloper commented on src/server/routes/jobsRoutes.js:
```diff
@@ -0,0 +1,40 @@
+//   }
+// })
+router.get('/', jobsControllers.getAllJobs)
+
+//returns all information of a specific job_id
+// router.get('/:jobId', async (req, res) => {
+
+//     try{
+//         const job = await Job.findOne({job_id: req.params.jobId});
+//         res.send(job);
+//     }catch(err){
+//         res.status[400].json({message: "Something went wrong."})
+//     }
+  
+// })
+
```
They were used for debugging. I'll remove them now. Thanks for the spotting.

DasDeveloper commented:
> it looks good overall. i guess we are using some sort of a predefined id to make the request to the backend ?We will probably have to change this in the future and have it fetch the id of the logged in recruiter.

Yes that's correct.

DasDeveloper committed:
Removed unnecessary comments

DasDeveloper committed:
Added a scrollbar for job posts when it overflows

DasDeveloper committed:
Fixed issues related to addJobAppliedToUser

Fixed the issue where the backend was unable to change the attribute jobsApplied in the database.

Kunal22shah reviewed (approved):
LGTM, if you can get the tests working with ~80% coverage it'd be great

DasDeveloper committed:
Update usersController.js

DasDeveloper committed:
Merge branch 'dev' into UserStoryIssue#32

Aboudik1452 commented:
Good work!
